coldfront/plugins/qumulo/tasks.py

```diff
@@ -97,6 +102,52 @@ def ingest_quotas_with_daily_usage() -> None:
     __validate_results(quota_usages, logger)
 
 
+def addUsersToADGroup(
+    wustlkeys: list[str], acl_allocation: Allocation, bad_keys: list = []
+) -> None:
+    if len(wustlkeys) == 0:
+        if len(bad_keys) > 0:
+            __send_invalid_users_email(acl_allocation, bad_keys)
+        return
+
+    wustlkey = wustlkeys[0]
+    group_name = acl_allocation.get_attribute("storage_acl_name")
+    active_directory_api = ActiveDirectoryAPI()
+
+    success = False
+    try:
+        active_directory_api.get_user(wustlkey)
+        success = True
+    except ValueError:
+        bad_keys.append(wustlkey)
+        success = False
+
+    if success:
+        active_directory_api.add_user_to_ad_group(wustlkey, group_name)
+        AclAllocations.add_user_to_access_allocation(wustlkey, acl_allocation)
+
+    async_task(addUsersToADGroup, wustlkeys[1:], acl_allocation, bad_keys)
+
+
+def __send_invalid_users_email(acl_allocation, bad_keys):
+    ctx = email_template_context()
+
+    CENTER_BASE_URL = import_from_settings("CENTER_BASE_URL")
+    ctx["allocation_url"] = f"{CENTER_BASE_URL}/allocation/{acl_allocation.id}"
+    ctx["access_type"] = (
+        "Read Only" if acl_allocation.resources.first().name == "ro" else "Read Write"
+    )
+    ctx["invalid_users"] = bad_keys
+
+    send_email_template(
+        subject="Users not found in Storage Allocation",
+        template_name="email/invalid_users.txt",
+        template_context=ctx,
+        sender=import_from_settings("DEFAULT_FROM_EMAIL"),
+        receiver_list=[],  # TODO: add receiver list
```

Comment:
@gunnarwustl has a PR that is defining a group of users in the `User` collection.  Could this simply retrieve the emails of the users in that group and use that as the receiver list?

Reply:
In fact, it's merged (https://github.com/WashU-IT-RIS/coldfront-wustl-fork/pull/16/files).  So would the `RIS-UserSupport` work for your purpose here?

Reply:
Yeah, I think so